dc-cudami-server/dc-cudami-server-business/src/main/java/de/digitalcollections/cudami/server/business/impl/service/identifiable/entity/parts/WebpageServiceImpl.java

```diff
@@ -75,6 +76,18 @@ public Webpage getParent(UUID webpageUuid) {
     return (Webpage) ((WebpageRepository) repository).getParent(webpageUuid);
   }
 
+  @Override
+  public Website getWebsite(UUID webpageUuid) {
+    UUID rootWebpageUuid = webpageUuid;
+    Webpage parent = getParent(webpageUuid);
+    while (parent != null) {
+      rootWebpageUuid = parent.getUuid();
+      parent = getParent(parent);
+    }
```

Comment:
Das ist ja son N+1 Datenbankquery Anti-Pattern, geht das nicht direkt im SQL?

Reply:
Stichwort hier: Common Tree Expressions (CTE) für hierarchische SQL queries.

Reply:
Ausgelagert in https://jira.digitale-sammlungen.de/browse/MR-960